database.py

```
@@ -0,0 +1,63 @@
+        try:
+            print(f"Attempting to connect to MongoDB at {MONGO_URI}")
+            client = MongoClient(MONGO_URI, serverSelectionTimeoutMS=5000)
+            # The ismaster command is cheap and does not require auth.
+            client.admin.command('ismaster')
+            db = client[DB_NAME]
+            print(f"Successfully connected to MongoDB. Using database: {DB_NAME}")
+        except ConnectionFailure as e:
+            print(f"MongoDB connection failed: {e}")
+            client = None
+            db = None
+            raise ConnectionFailure(f"Could not connect to MongoDB: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)